cluster-node-image-builder container fails in Azure pipeline

**What steps did you take and what happened:**

When running Azure image-builder tasks from this repository in pipelines, we want to use the container produced by `make -C images/capi docker-build` as recommended in the [Image Builder book](https://image-builder.sigs.k8s.io/capi/container-image.html#examples). But in practice it errors out, as shown in the below log.

This same startup script with `useradd` runs successfully when backed by the old [deis/go-dev](https://github.com/deis/docker-go-dev) container.

**What did you expect to happen:**


**Anything else you would like to add:**

```
Starting: Initialize containers
/usr/bin/docker version --format '{{.Server.APIVersion}}'
'1.41'
Docker daemon API version: '1.41'
/usr/bin/docker version --format '{{.Client.APIVersion}}'
'1.41'
Docker client API version: '1.41'
/usr/bin/docker ps --all --quiet --no-trunc --filter "label=d03b08"
/usr/bin/docker network prune --force --filter "label=d03b08"
/usr/bin/docker pull k8s.gcr.io/scl-image-builder/cluster-node-image-builder-amd64:v0.1.13
v0.1.13: Pulling from scl-image-builder/cluster-node-image-builder-amd64
675920708c8b: Pulling fs layer
...
b955c67dff1f: Pull complete
Digest: sha256:cd095a2718006fffeced8acdf296e65b108999e1c742672d6dea83ee45c6e060
Status: Downloaded newer image for k8s.gcr.io/scl-image-builder/cluster-node-image-builder-amd64:v0.1.13
k8s.gcr.io/scl-image-builder/cluster-node-image-builder-amd64:v0.1.13
/usr/bin/docker info -f "{{range .Plugins.Network}}{{println .}}{{end}}"
bridge
host
ipvlan
macvlan
null
overlay
/usr/bin/docker network create --label d03b08 vsts_network_14e080230a5f4d178a8f7af4a7ce14d9
063dbf403c87b9cf2d4fef33f83d6c772495242ce91f2baf87fe13b254efacb7
/usr/bin/docker inspect --format="{{index .Config.Labels \"com.azure.dev.pipelines.agent.handler.node.path\"}}" k8s.gcr.io/scl-image-builder/cluster-node-image-builder-amd64:v0.1.13
/usr/bin/docker create --name dbeddeee79b04fad83f8ca3caadbcedb_k8sgcriosclimagebuilderclusternodeimagebuilderamd64v0113_ab9354 --label d03b08 --network vsts_network_14e080230a5f4d178a8f7af4a7ce14d9  -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/mnt/vss/_work/1":"/__w/1" -v "/mnt/vss/_work/_temp":"/__w/_temp" -v "/mnt/vss/_work/_tasks":"/__w/_tasks" -v "/mnt/vss/_work/_tool":"/__t" -v "/usr/local/vss-agent/2.211.0/externals":"/__a/externals":ro -v "/mnt/vss/_work/.taskkey":"/__w/.taskkey" k8s.gcr.io/scl-image-builder/cluster-node-image-builder-amd64:v0.1.13 "/__a/externals/node/bin/node" -e "setInterval(function(){}, 24 * 60 * 60 * 1000);"
de54b37a05ca92436abb12da1038bfc8efd01a1aed922c70459375e7085cddcf
/usr/bin/docker start de54b37a05ca92436abb12da1038bfc8efd01a1aed922c70459375e7085cddcf
de54b37a05ca92436abb12da1038bfc8efd01a1aed922c70459375e7085cddcf
/usr/bin/docker ps --all --filter id=de54b37a05ca92436abb12da1038bfc8efd01a1aed922c70459375e7085cddcf --filter status=running --no-trunc --format "{{.ID}} {{.Status}}"
de54b37a05ca92436abb12da1038bfc8efd01a1aed922c70459375e7085cddcf Up Less than a second
/usr/bin/docker exec  de54b37a05ca92436abb12da1038bfc8efd01a1aed922c70459375e7085cddcf sh -c "command -v bash"
whoami 
cloudtest
id -u cloudtest
1001
id -g cloudtest
1001
id -gn cloudtest
cloudtest
Try to create a user with UID '1001' inside the container.
/usr/bin/docker exec  de54b37a05ca92436abb12da1038bfc8efd01a1aed922c70459375e7085cddcf bash -c "getent passwd 1001 | cut -d: -f1 "
/usr/bin/docker exec  de54b37a05ca92436abb12da1038bfc8efd01a1aed922c70459375e7085cddcf useradd -m -u 1001 cloudtest_azpcontainer
##[error]Docker-exec executed: `useradd -m -u 1001 cloudtest_azpcontainer`; container id: `de54b37a05ca92436abb12da1038bfc8efd01a1aed922c70459375e7085cddcf`; exit code: `1`; command output: `useradd: Permission denied.`, `useradd: cannot lock /etc/passwd; try again later.`
Finishing: Initialize containers
```


**Environment:**

Project ([Image Builder for Cluster API](https://github.com/kubernetes-sigs/image-builder/tree/master/images/capi), [kube-deploy/imagebuilder](https://github.com/kubernetes-sigs/image-builder/tree/master/images/kube-deploy/imagebuilder), [konfigadm](https://github.com/kubernetes-sigs/image-builder/tree/master/images/konfigadm)):

Additional info for Image Builder for Cluster API related issues:

- OS (e.g. from `/etc/os-release`, or `cmd /c ver`):
- Packer Version:
- Packer Provider:
- Ansible Version: 
- Cluster-api version (if using):
- Kubernetes version: (use `kubectl version`):

/kind bug

--- a/images/capi/packer/azure/.pipelines/build-vhd.yaml
+++ b/images/capi/packer/azure/.pipelines/build-vhd.yaml
@@ -1,6 +1,5 @@
 # Required pipeline variables:
 # - BUILD_POOL - Azure DevOps build pool to use
-# - CONTAINER_IMAGE - Dev container image URL to use. Should have Azure CLI, Packer and Ansible.
 # - AZURE_TENANT_ID - tenant ID
 # - AZURE_CLIENT_ID - Service principal ID
 # - AZURE_CLIENT_SECRET - Service principal secret
@@ -11,7 +10,6 @@
 
 jobs:
 - job: build_vhd
-  container: $[ variables['CONTAINER_IMAGE'] ]
   timeoutInMinutes: 120
   strategy:
     maxParallel: 0
@@ -34,6 +32,7 @@ jobs:
     condition: and(eq(variables['PREFLIGHT_CHECKS'], 'true'), eq(variables['OS'], 'Windows'))
   - script: |
       set -o pipefail
+      export PATH=$PATH:$HOME/.local/bin LC_ALL=en_US.UTF-8
       make deps-azure
       os=$(echo "${OS}" | tr '[:upper:]' '[:lower:]')
       version=$(echo "${OS_VERSION}" | tr '[:upper:]' '[:lower:]' | tr -d .)

--- a/images/capi/packer/azure/.pipelines/cleanup.yaml
+++ b/images/capi/packer/azure/.pipelines/cleanup.yaml
@@ -1,6 +1,5 @@
 # Required pipeline variables:
 # - BUILD_POOL - Azure DevOps build pool to use
-# - CONTAINER_IMAGE - Dev container image URL to use. Should have Azure CLI, Packer and Ansible.
 # - AZURE_CLIENT_ID_SKU - Service principal ID to PUT the SKU
 # - AZURE_CLIENT_SECRET_SKU - Service principal secret to PUT the SKU
 # - AZURE_TENANT_ID_SKU - tenant ID to PUT the SKU
@@ -27,7 +26,6 @@ stages:
   - stage: clean
     jobs:
     - job:
-      container: $[ variables['CONTAINER_IMAGE'] ]
       timeoutInMinutes: 120
       pool:
         name: $(BUILD_POOL)

--- a/images/capi/packer/azure/.pipelines/create-disk-version.yaml
+++ b/images/capi/packer/azure/.pipelines/create-disk-version.yaml
@@ -1,6 +1,5 @@
 # Required pipeline variables:
 # - BUILD_POOL - Azure DevOps build pool to use
-# - CONTAINER_IMAGE - Dev container image URL to use. Should have Azure CLI, Packer and Ansible.
 # - AZURE_TENANT_ID - tenant ID
 # - AZURE_CLIENT_ID - Service principal ID
 # - AZURE_CLIENT_SECRET - Service principal secret
@@ -9,7 +8,6 @@
 
 jobs:
 - job: create_disk_version
-  container: $[ variables['CONTAINER_IMAGE'] ]
   timeoutInMinutes: 120
   strategy:
     maxParallel: 0

--- a/images/capi/packer/azure/.pipelines/create-sku.yaml
+++ b/images/capi/packer/azure/.pipelines/create-sku.yaml
@@ -1,6 +1,5 @@
 # Required pipeline variables:
 # - BUILD_POOL - Azure DevOps build pool to use
-# - CONTAINER_IMAGE - Dev container image URL to use. Should have Azure CLI, Packer and Ansible.
 # - AZURE_CLIENT_ID - Service principal ID
 # - AZURE_CLIENT_SECRET - Service principal secret
 # - AZURE_TENANT_ID - tenant ID
@@ -14,7 +13,6 @@
 
 jobs:
 - job: create_sku
-  container: $[ variables['CONTAINER_IMAGE'] ]
   timeoutInMinutes: 120
   strategy:
     maxParallel: 0

--- a/images/capi/packer/azure/.pipelines/smoke-test.yaml
+++ b/images/capi/packer/azure/.pipelines/smoke-test.yaml
@@ -1,6 +1,5 @@
 # Required pipeline variables:
 # - BUILD_POOL - Azure DevOps build pool to use
-# - CONTAINER_IMAGE - Dev container image URL to use. Should have Azure CLI, Packer and Ansible.
 # - AZURE_TENANT_ID_VHD - tenant ID to build the vhd
 # - AZURE_CLIENT_ID_VHD - Service principal ID to build the vhd
 # - AZURE_CLIENT_SECRET_VHD - Service principal secret to build the vhd
@@ -25,7 +24,6 @@ stages:
   - stage: vhd
     jobs:
     - job:
-      container: $[ variables['CONTAINER_IMAGE'] ]
       timeoutInMinutes: 120
       pool:
         name: $(BUILD_POOL)
@@ -33,6 +31,7 @@ stages:
       - template: k8s-config.yaml
       - script: |
           set -o pipefail
+          export PATH=$PATH:$HOME/.local/bin LC_ALL=en_US.UTF-8
           make deps-azure
           os=$(echo "$OS" | tr '[:upper:]' '[:lower:]')
           version=$(echo "$OS_VERSION" | tr '[:upper:]' '[:lower:]' | tr -d .)

--- a/images/capi/packer/azure/.pipelines/stages.yaml
+++ b/images/capi/packer/azure/.pipelines/stages.yaml
@@ -1,6 +1,5 @@
 # Required pipeline variables:
 # - BUILD_POOL - Azure DevOps build pool to use
-# - CONTAINER_IMAGE - Dev container image URL to use. Should have Azure CLI, Packer and Ansible.
 # - AZURE_TENANT_ID_VHD - tenant ID to build the vhd
 # - AZURE_CLIENT_ID_VHD - Service principal ID to build the vhd
 # - AZURE_CLIENT_SECRET_VHD - Service principal secret to build the vhd

--- a/images/capi/packer/azure/.pipelines/test-vhd.yaml
+++ b/images/capi/packer/azure/.pipelines/test-vhd.yaml
@@ -1,6 +1,5 @@
 # Required pipeline variables:
 # - BUILD_POOL - Azure DevOps build pool to use
-# - CONTAINER_IMAGE - Dev container image URL to use. Should have Azure CLI, Packer, and Ansible.
 # - AZ_CAPI_EXTENSION_URL - URL to the Azure CAPI extension build.
 # - AZURE_TENANT_ID - tenant ID
 # - AZURE_CLIENT_ID - Service principal ID
@@ -12,7 +11,6 @@
 
 jobs:
 - job: test_vhd
-  container: $[ variables['CONTAINER_IMAGE'] ]
   timeoutInMinutes: 120
   strategy:
     maxParallel: 0
@@ -104,7 +102,6 @@ jobs:
       # Set up the Azure CLI Cluster API extension
       # https://github.com/Azure/azure-capi-cli-extension/releases/download/az-capi-nightly/capi-0.0.vnext-py2.py3-none-any.whl
       python3 -m pip install --upgrade pip
-      az upgrade --yes
       az extension add --yes --source "${AZ_CAPI_EXTENSION_URL}"
 
       # Install required binaries

